### PR TITLE
git commit policy

### PR DESCRIPTION
**Pull request by joejstuart**
warn on non-conforming commit message

### CODE REVIEW

File: policy/release/commit_signature_test.rego [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package policy.release.commit_signature
+
+import data.lib
+
+mock_data(signatures) = d {
+    d := {"body": {"Hash": ["nm32n5235"]},"signatures": signatures}
```
Comment (zregvart) [inline policy/release/commit_signature_test.rego:6]: (nit) the structure could be simpler, something like:

```json
{
  "<commit-id1>": ["email1", "email2", ...],
  "<commit-id2>": ["email3", "email4", ...],
}
```

**Review by zregvart — APPROVED** [review]
Pragmatic enough for a proof of concept. Run `make fmt` to pass the formatting lint check.

File: policy/release/commit_signature.rego [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#     - redhat.com
+warn[result] {
+    signature := input.signatures[_]
+    domain := split(signature, "@")
+    not known_domains(domain[1], rego.metadata.rule().custom.rule_data.allowed_email_domains)
+    result := lib.result_helper(rego.metadata.chain(), [signature, input.body.Hash])
+}
+
+known_domains(domain, allowed_email_domains) {
+    lib.item_in_list(domain, allowed_email_domains)
```
Comment (lcarva) [inline policy/release/commit_signature.rego:39]: Might be worth making this check case insensitive.

File: policy/release/commit_signature_test.rego (hunk 1 above)
Comment (lcarva) [reply policy/release/commit_signature_test.rego:6]: Just for my own understanding, why is `Hash` an array instead of a single string?

**Review by lcarva — APPROVED** [review]

**Pull request merged**